packages/api/src/utils/stripe.js

```diff
@@ -256,7 +258,26 @@ export class StripeCustomersService {
       name: options?.name,
       email: options?.email
     })
-    await this.userCustomerService.upsertUserCustomer(user.id.toString(), createdCustomer.id)
+
+    try {
+      await this.userCustomerService.insertUserCustomer(user.id.toString(), createdCustomer.id)
+    } catch (error) {
+      // due to potential race conditions, we want to ensure we only return an actual associated stripe customer
+      // and remove stripe customers that we fail to associate
+      if (hasOwnProperty(error, 'code') && error.code === 'CONSTRAINT_ERROR_DB') {
+        // no need to block during deletion
+        this.stripe.customers.del(createdCustomer.id)
+
+        existingCustomer = await this.userCustomerService.getUserCustomer(user.id.toString())
+
+        if (!existingCustomer) {
+          throw new CustomerNotFoundError()
```

Comment:
... this should never happen right?

is there a test case that necessitates it? what should the caller of this func do if they encounter `CustomerNotFoundError`?

it seems like maybe this should never actually happen... and if it did.... it would be pretty surprising.... like maybe all this should just be `throw error` here and no need for CustomerNotFoundError which to me looks like it means `UnableToCreateCustomerForUser` or something (because you're throwing this in the context of encountering an error doing insertUserCustomer)